### ISSUE TITLE
Deploy to Mobile Center returning EISDIR: illegal operation on a directory, read

### ISSUE THREAD

**Issue by bojanmisic** [Microsoft/vsts-tasks#4918]
I'm using "Deploy to Visual Studio Mobile Center" task in VSTS to deploy binary to Mobile Center. I have setup API access from VSTS to Mobile Center with the API key from Mobile Center with "Full Access". I have set up App Slug correctly so I'm not sure what is the problem.

Here is the task output:

```
2017-07-31T13:04:07.3321637Z ##[section]Starting: Deploy to Visual Studio Mobile Center
2017-07-31T13:04:07.3321637Z ==============================================================================
2017-07-31T13:04:07.3321637Z Task         : Mobile Center Upload
2017-07-31T13:04:07.3321637Z Description  : Upload mobile app packages to Visual Studio Mobile Center
2017-07-31T13:04:07.3321637Z Version      : 0.119.1
2017-07-31T13:04:07.3321637Z Author       : Microsoft Corporation
2017-07-31T13:04:07.3331638Z Help         : Upload mobile app packages to Visual Studio Mobile Center
2017-07-31T13:04:07.3331638Z ==============================================================================
2017-07-31T13:04:09.1952666Z ##[error]Error: form-data: EISDIR: illegal operation on a directory, read
2017-07-31T13:04:09.3374890Z ##[section]Finishing: Deploy to Visual Studio Mobile Center
```

Please ask if you need any more info.

Thanks,
Bojan

**Comment by bojanmisic** [comment]
User mistake, I apologize.

**Comment by joost4net** [comment]
Hi Bojanmisic, what was the problem? I'm currently having the same issue. I'll tried a couple of times with different api keys with all full access. 
Path to publish I used $(build.artifactstagingdirectory) or $(agent.builddirectory)/output/$(SDK)/$(Configuration) or $(SDK)/$(Configuration)/build.sym/$(Configuration)-$(SDK)/output. But on all three folders I got the same error. I'm trying to deploy an ipa file by the way. And I'm doing this after the copy statement

**Comment by wattsrw** [comment]
@joost4net 
I had the same error message. The fix for me was to put the file name at the end of the binary file path.
Our binary path is:
$(Build.SourcesDirectory)\bin\$(BuildConfiguration)\$(AndroidProjectName).apk
So it just depends where you're outputting/copying your apk and ipa files, just be sure to have the file name at the end of the path.

**Comment by fldvrees** [comment]
I had the same error message.

The error was on my side. In the build task I selected `Select Release Notes File` without specifying a valid `Release notes file`.

To be honest: The error message could be better. I lost about 20 minutes finding the source of this error...

**Comment by SuperDuperOne** [comment]
@bojanmisic agreed, it would be help as we're having the same issue here.

**Comment by bojanmisic** [comment]
It's a dummy error message which can be related to a number of things. I admit I could have been a better open source contributor and write what was the original issue, but at that time it was a time pressured project and a really silly mistake which I can't remember at the moment :(. Apologies to everyone on the thread, but I agree with the above - the error message could be better!

**Comment by ColinDabritzQ5id** [comment]
TLDR: Make sure you are using the "File" form of the parameters, and paths provided are generally correct.

In case this helps anyone else: EISDIR means "E IS DIR" = "Error: Is Directory".

Per the docs:
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/app-center-distribute?view=azure-devops

releaseNotesOption: 'input' # Options: input, file
releaseNotesInput: # Required when releaseNotesOption == Input
releaseNotesFile: # Required when releaseNotesOption == File

We had a step set up with "input" but it was useless, so we ported over a process from another build that pulled out the latest git commit message and put it into a file, which was then used to set the releaseNotes field.

We had it set up for input before, and we provided a full file path now, so I switched over the releaseNotesOption to 'file'. But I did NOT change the name of the other parameter, and was passing my file path into the 'releaseNotesInput' parameter instead. Which was ignored, and which means the implicit "blank" was provided for the releaseNotesFile parameter because it was unspecified.

Which probably meant that the script attempted to read my file, as a file. But because it was blank, something that was expected a path like "C:\mysource\mynotesfile.md" instead got a blank file name, so it attempted to open "C:\mysource\" AS A FILE, and thus the error above, you can't read a path that is a directory as if it were a file.

The error messaging could be better.

Other issues with paths could potentially cause this error, if a file is expected, but the path is a directory instead. It may be helpful for troubleshooting to add a script just before this step that prints out all the paths that are used to verify they exist and are files or directories as you expect them to be.

Stay safe out there!
